woocommerce/woocommerce-android
# Add analytics to hub settings

Closes: #10704 

### Why
As part of Expanded Analytics Hub milestone 2, we plan to add support for customizing the cards we display on the Analytic Hub screen.

### Description
This PR adds track events to the new Customize analytics screen.

 ### New events

 `analytics_hub_settings_opened` Registration: https://github.com/Automattic/tracks-events-registration/pull/2262
Triggered when the user taps on the settings icon in the Analytics Hub, to open the settings screen.}

`analytics_hub_settings_saved` Registration: https://github.com/Automattic/tracks-events-registration/pull/2263
Triggered when the user taps on the save menu to save the configuration
Custom prop `enabled_cards `with value set to a comma-separated list of all the currently enabled cards (e.g. revenue,orders,products)
Custom prop `disabled_cards` with value set to a comma-separated list of all the currently disabled cards
 

### Testing instructions
TC1
1. Open the app
2. Tap on See more
3. Tap on the pencil icon to navigate to the Customize Analytics screen
4. Check that the `analytics_hub_settings_opened` event is triggered

TC2
1. Open the app
2. Tap on See more
3. Tap on the pencil icon to navigate to the Customize Analytics screen
4. Edit the configuration to enable the Save button
5. Tap on Save
6. Check that the `analytics_hub_settings_saved` is triggered
7. Check that the `enabled_cards` property is part of the event, and the list of enabled cards matches the current configuration
8. Check that the `disabled_cards` property is part of the event, and the list of disabled cards matches the current configuration

### Images/gif
```
🔵 Tracked: analytics_hub_settings_opened, Properties: {"blog_id":214253715,"is_wpcom_store":true,"was_ecommerce_trial":false,"plan_product_slug":"ecommerce-bundle","store_id":"78fb3abe-869e-4280-8dcf-c4c94c9b84b6","is_debug":true,"site_url":"https:\/\/superlativecentaur.wpcomstaging.com"}

🔵 Tracked: analytics_hub_settings_saved, Properties: {"enabled_cards":"session,orders","disabled_cards":"revenue,products","blog_id":214253715,"is_wpcom_store":true,"was_ecommerce_trial":false,"plan_product_slug":"ecommerce-bundle","store_id":"78fb3abe-869e-4280-8dcf-c4c94c9b84b6","is_debug":true,"site_url":"https:\/\/superlativecentaur.wpcomstaging.com"}
```
